testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -858,53 +863,42 @@ def test_segid_and_resid():
+    def _check_sels(self, ref, sel, universe):
+        ref = universe.select_atoms(ref)
+        sel = universe.select_atoms(sel)
 
         assert_array_equal(ref.indices, sel.indices)
 
-    def test_string_selections(self):
-        for ref, sel in (
-                ('name NameABA or name NameACA or name NameADA',
-                 'name NameABA NameACA NameADA'),
-                ('type TypeE or type TypeD or type TypeB',
-                 'type TypeE TypeD TypeB'),
-                ('resname RsC or resname RsY', 'resname RsC RsY'),
-                ('name NameAB* or name NameACC', 'name NameAB* NameACC'),
-                ('(name NameABC or name NameABB) and (resname RsD or resname RsF)',
-                 'name NameABC NameABB and resname RsD RsF'),
-                ('segid SegA or segid SegC', 'segid SegA SegC'),
-        ):
-            yield self._check_sels, ref, sel
-
-    def test_range_selections(self):
-        # All these selections just use numeric types,
-        # So loop over what type of selections,
-        # And apply the same numeric constraints to all
-        for seltype in ['resid', 'resnum', 'bynum']:
-            for ref, sel in (
-                    ('{typ} 1 or {typ} 2', '{typ} 1 2'),
-                    ('{typ} 1:10 or {typ} 22', '{typ} 1:10 22'),
-                    ('{typ} 1:10 or {typ} 20:30', '{typ} 1:10 20:30'),
-                    ('{typ} 1-5 or {typ} 7', '{typ} 1-5 7'),
-                    ('{typ} 1-5 or {typ} 7:10 or {typ} 12',
-                     '{typ} 1-5 7:10 12'),
-                    ('{typ} 1 or {typ} 3 or {typ} 5:10', '{typ} 1 3 5:10'),
-            ):
-                yield (self._check_sels,
-                       ref.format(typ=seltype),
-                       sel.format(typ=seltype))
-
-class TestICodeSelection(object):
+    @pytest.mark.parametrize('ref, sel',[
+        ('name NameABA or name NameACA or name NameADA', 'name NameABA NameACA NameADA'),
+        ('type TypeE or type TypeD or type TypeB', 'type TypeE TypeD TypeB'),
+        ('resname RsC or resname RsY', 'resname RsC RsY'),
+        ('name NameAB* or name NameACC', 'name NameAB* NameACC'),
+        ('segid SegA or segid SegC', 'segid SegA SegC'),
+        ('(name NameABC or name NameABB) and (resname RsD or resname RsF)', 'name NameABC NameABB and resname RsD RsF'),
+    ])
+    def test_string_selections(self, ref, sel, universe):
+        self._check_sels(ref, sel, universe)
+
+    @pytest.mark.parametrize("seltype", ['resid', 'resnum', 'bynum'])
+    @pytest.mark.parametrize('ref, sel', [
+        ('{typ} 1 or {typ} 2', '{typ} 1 2'),
+        ('{typ} 1:10 or {typ} 22', '{typ} 1:10 22'),
+        ('{typ} 1:10 or {typ} 20:30', '{typ} 1:10 20:30'),
+        ('{typ} 1-5 or {typ} 7', '{typ} 1-5 7'),
+        ('{typ} 1-5 or {typ} 7:10 or {typ} 12', '{typ} 1-5 7:10 12'),
+        ('{typ} 1 or {typ} 3 or {typ} 5:10', '{typ} 1 3 5:10'),
+    ])
+    def test_range_selections(self, seltype, ref, sel, universe):
```

Comment:
how many tests are generated for this? Does it equal the number of tests the old double loop used in nose?